File: Restaurant-Manager-Java/src/models/ProductOrder.java
package models;

import java.math.BigDecimal;

public class ProductOrder {
    private final FoodProduct _product;
    private final int _quantity;
    private final BigDecimal _totalPrice;

    public ProductOrder(FoodProduct product, int quantity) {
        _product = product;
        _quantity = quantity;
        _totalPrice = product.getPrice().multiply(BigDecimal.valueOf(quantity));
    }

    public FoodProduct getProduct() {
        return _product;
    }

    public int getQuantity() {
        return _quantity;
    }

    public BigDecimal getTotalPrice() {
        return _totalPrice;
    }
}

File: Restaurant-Manager-C#/Restaurant-Manager.NET/Models/ProductOrder.cs
namespace CollectionsLinq.Models
{
    public class ProductOrder
    {
        public ProductOrder(FoodProduct product, int quantity)
        {
            this.Product = product;
            this.Quantity = quantity;
            this.TotalPrice = product.Price * quantity;
        }

        public FoodProduct Product { get; }

        public int Quantity { get; }

        public decimal TotalPrice { get; }
    }
}
